### PR TITLE
Improve GPU debug reports

### PR DESCRIPTION
**Pull request by System64fumo**
#### Describe your PR, what does it fix/add?
Instead of using weird "hacks" to get GPU information, Just get whatever libdrm reports.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
At the moment this uses /dev/dri/card0.
This implementation is not as verbose as before, Card model is not shown, unaware of a way to show it.

#### Is it ready for merging, or does it need work?
I'd say it's ready i guess.

### CODE REVIEW

**Comment by MahouShoujoMivutilde** [comment]
More GPU info can be obtained like so https://gitlab.freedesktop.org/mesa/demos/-/blob/main/src/demos/glinfo.c?ref_type=heads#L23-25

```
GL_VERSION: 4.6.0 NVIDIA 560.31.02
GL_RENDERER: NVIDIA GeForce GTX 970/PCIe/SSE2
GL_VENDOR: NVIDIA Corporation
GL_EXTENSIONS: blah blah blah long string
GL_SHADING_LANGUAGE_VERSION = 4.60 NVIDIA
GLU_VERSION: 1.3
GLU_EXTENSIONS: GLU_EXT_nurbs_tessellator GLU_EXT_object_space_tess 
GLUT_API_VERSION: 4
GLUT_XLIB_IMPLEMENTATION: 13
```

You also need to run drmFreeVersion() after you done with it.

https://github.com/swaywm/sway/blob/7288f77bbe275825a0e0b011db873d9367782af0/sway/server.c#L148-L175

**Comment by System64fumo** [comment]
Yup sorry, Completely forgot to include that, As for additional GPU info vaxry said the current output is fine.
At the end of the day this is not a fetch program, Specifics shouldn't matter i guess.

*This is just an excuse, I'm too lazy to add more crud*

**Comment by vaxerski** [comment]
y no clang-format

**Comment by System64fumo** [comment]
Didn't know that was a thing?

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -920,14 +923,25 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
 
-#if defined(__DragonFly__) || defined(__FreeBSD__)
-    const std::string GPUINFO = execAndGet("pciconf -lv | fgrep -A4 vga");
-#elif defined(__arm__) || defined(__aarch64__)
-    const std::string GPUINFO = execAndGet("cat /proc/device-tree/soc*/gpu*/compatible");
-#else
-    const std::string GPUINFO = execAndGet("lspci -vnn | grep VGA");
-#endif
-    result += "GPU information: \n" + GPUINFO;
+    int fd = open("/dev/dri/card0", O_RDONLY | O_CLOEXEC);
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:926]: this is dumb, we may not even be running on card0. You should get the card(s) path(s) from the current aq session g_pCompositor->m_pAqBackend->session

File: src/debug/HyprCtl.cpp [hunk 2]
```diff
@@ -920,14 +923,25 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
+    if (fd < 0) {
+        throw std::runtime_error("Failed to open /dev/dri/card0");
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:928]: ??? just skip this log

File: src/debug/HyprCtl.cpp [hunk 3]
```diff
@@ -920,14 +923,25 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
+    }
+    drmVersion* version = drmGetVersion(fd);
+    if (!version) {
+        close(fd);
+        throw std::runtime_error("Failed to get DRM version");
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:933]: again

File: src/helpers/MiscFunctions.cpp [hunk 4]
```diff
@@ -609,14 +611,24 @@ void logSystemInfo() {
...
     Debug::log(NONE, "\n");
 
-#if defined(__DragonFly__) || defined(__FreeBSD__)
-    const std::string GPUINFO = execAndGet("pciconf -lv | fgrep -A4 vga");
-#elif defined(__arm__) || defined(__aarch64__)
-    const std::string GPUINFO = execAndGet("cat /proc/device-tree/soc*/gpu*/compatible");
-#else
-    const std::string GPUINFO = execAndGet("lspci -vnn | grep VGA");
-#endif
+    int fd = open("/dev/dri/card0", O_RDONLY | O_CLOEXEC);
```
Comment (vaxerski) [inline src/helpers/MiscFunctions.cpp:614]: same stuff here, also why are you duping code? make this into a function? renderer does the same, btw, idk why this is needed here

**Comment by System64fumo** [comment]
No clue how to use "g_pCompositor->m_pAqBackend->session" but hopefully this is better.

**Comment by MahouShoujoMivutilde** [comment]
Probably like so https://github.com/hyprwm/Hyprland/blob/688fe5c14781c63a1db23d4d02bf239283068ff6/src/render/Renderer.cpp#L38

**Pull request closed without merging**